MyApp/app/controllers/team_controller.rb

```diff
@@ -1,9 +1,13 @@
 class TeamController < ApplicationController
 	before_action :authenticate_user!
+	before_action :admin?, only: [:index]
+
+  def admin?
+    redirect_to root_path if current_user.role != 'admin'
```

Comment:
It can be done here and its ok, but for future try to use pundit policies, its easier to have every access point on the same place